test/e2e/tests/set_multiple_jobs.py

```diff
@@ -0,0 +1,129 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import json
+import logging
+import sys
+
+
+SLOW = False
+
+
+def get_init_args():
+    return {
+        "kwargs": {
+            "project": "foo",
+        }
+    }
+
+
+def get_jobs():
+    return []
+
+
+def get_set_jobs_args():
+    jobs = [{
```

Comment:
OK, this highlights something I hadn't noticed before.

It seems like it's necessary for the user to supply *all* of these keys, right? They can't for example omit `"outputs"` even if the value is `None`?

That seems (1) unergonomic and (2) we would break everybody's scripts every time we add a new key to the schema, because if somebody's script is doing `set-jobs` with a certain set of keys that doesn't include the new key that we just introduced, then something somewhere will crash.

I think every flag that has a default or optional value in the command line needs to also be that way in the JSON. Namely the only required keys in the set-jobs dict should be command, pipeline, ci_stage.

I realise that this will make the implementation a lot more complicated :'( But I don't think we can merge this in like this.